### PR TITLE
Refactor test_snmpserver

### PR DESCRIPTION
**Pull request by chrisvanheuveln**
- Shortened class/testcase names
- Removed redundancy by combining several test sections into a single test
- Added validate_property checks
- Tested on n9-i4

edit: just realized I used the wrong filename in my title. These changes are all in the test_snmpserver.rb file.

### CODE REVIEW

File: tests/test_snmpserver.rb [hunk 1]
```diff
@@ -46,113 +46,55 @@ def setup
...
 
-  def test_snmpserver_aaa_user_cache_timeout_set_invalid_upper_range
-    return if platform != :nexus
-    snmpserver = SnmpServer.new
-    assert_raises(Cisco::CliError) do
-      snmpserver.aaa_user_cache_timeout = 86_401
+  def test_aaa_user_cache_timeout
+    if validate_property_excluded?('snmp_server', 'aaa_user_cache_timeout')
+      assert_raises(Cisco::UnsupportedError) do
+        SnmpServer.new.aaa_user_cache_timeout = 1400
```
Comment (mikewiebe) [inline tests/test_snmpserver.rb]: Any particular reason for the timeout value change?

File: tests/test_snmpserver.rb [hunk 2]
```diff
@@ -46,113 +46,55 @@ def setup
...
+    assert_equal('', s.contact)
+
+    # negative test
+    assert_raises(TypeError) { s.contact = 123 }
   end
 
-  def test_snmpserver_sys_contact_set_special_chars
+  def test_contact_special_chars
     snmpserver = SnmpServer.new
     # newcontact = "Test{}(%tuvy@_cisco contact$#!@1234^&*()_+"
```
Comment (mikewiebe) [inline tests/test_snmpserver.rb]: Remove?

**Comment by mikewiebe** [comment]
@chrisvanheuveln 👍 Just one small suggestion to remove an already commented out line if possible.

File: tests/test_snmpserver.rb (hunk 2 above)
Comment (chrisvanheuveln) [inline tests/test_snmpserver.rb]: Sure. I assumed it was just another test input somebody wanted to keep around. I'll remove it.

**Pull request merged**